test-suite-data.json

```diff
@@ -47,6 +47,30 @@
     "subpath": "googleapis/api/annotations",
     "is_invalid": false
   },
+  {
+    "description": "invalid subpath - unencoded subpath cannot contain '..'",
+    "purl": "pkg:GOLANG/google.golang.org/genproto@abcdedf#/googleapis/%2E%2E/api/annotations/",
+    "canonical_purl": "pkg:golang/google.golang.org/genproto@abcdedf#googleapis/api/annotations",
+    "type": "golang",
+    "namespace": "google.golang.org",
+    "name": "genproto",
+    "version": "abcdedf",
+    "qualifiers": null,
+    "subpath": "googleapis/../api/annotations",
+    "is_invalid": true
```

Comment:
lets see how the different test scenarios behave:
- **passing** parsing the test canonical purl then re-building a purl from these parsed components should return the test canonical purl
- **failing, as the `..` is expected to be stripped** parsing the test purl should return the components parsed from the test canonical purl
- **failing, as the `..` is expected to be stripped** parsing the test purl then re-building a purl from these parsed components should return the test canonical purl
- **passing** building a purl from the test components should return the test canonical purl